src/dmd/astbase.d

```diff
@@ -3655,16 +3678,16 @@ struct ASTBase
         }
     }
 
-    extern (C++) final class TypeStruct : Type
+    extern (C++) final class TypeStruct : TypeAggregate
     {
-        StructDeclaration sym;
-        AliasThisRec att = AliasThisRec.fwdref;
         bool inuse = false;
 
+        extern (D) inout(StructDeclaration) sym() inout { return cast(typeof(return))asym; }
```

Comment:
@ibuclaw what about these? Do you have a better alternative?

Reply:
If the asym field is exposed, doesn't really matter to the C++ codegen. We can fix it up as necessary.

If you make the field private however, then these getters/setters must be extern(C++).